src/gui/wizard/selftest_frame_esp.cpp

```diff
@@ -0,0 +1,108 @@
+/**
+ * @file selftest_frame_esp.cpp
+ */
+
+#include "selftest_frame_esp.hpp"
+#include "selftest_esp_type.hpp"
+#include "i18n.h"
+#include "wizard_config.hpp"
+#include "marlin_client.hpp"
+#include "resource.h"
+
+static constexpr size_t icon_sz = 64;
+static constexpr size_t row_2 = WizardDefaults::row_1 + WizardDefaults::row_h + WizardDefaults::txt_h;
+static constexpr size_t row_2b = row_2 + WizardDefaults::progress_row_h;
+static constexpr size_t row_3 = row_2 + icon_sz;
+static constexpr size_t col_0 = WizardDefaults::MarginLeft;
+
+SelftestFrameESP::SelftestFrameESP(window_t *parent, PhasesSelftest ph, fsm::PhaseData data)
+    : AddSuperWindow<SelftestFrameWithRadio>(parent, ph, data)
+    , text_top(this, Rect16(col_0, WizardDefaults::row_1, WizardDefaults::X_space, WizardDefaults::txt_h * 2), is_multiline::yes)
+    , progress(this, row_2)
+    , text_progress(this, Rect16(col_0, row_2b, WizardDefaults::X_space, WizardDefaults::txt_h), is_multiline::no)
+    , icon(this, Rect16((GuiDefaults::ScreenWidth - icon_sz) / 2, row_2, icon_sz, icon_sz), IDR_PNG_wifi_large_64x64)
+
+    , text_bot(this, Rect16(col_0, row_3, WizardDefaults::X_space, WizardDefaults::txt_h * 4), is_multiline::yes)
+
+{
+    progress.SetProgressPercent(0);
+    text_top.SetAlignment(Align_t::LeftBottom());
+    text_bot.SetAlignment(Align_t::LeftCenter());
+    text_progress.SetAlignment(Align_t::Center());
+    change();
+}
+
+void SelftestFrameESP::change() {
+    SelftestESP_t dt(data_current);
+    progress.SetProgressPercent(dt.progress);
+
+    //"[0 / 0]";
+    progr_text[1] = dt.current_file + '0';
+    progr_text[5] = dt.count_of_files + '0';
+
+    const char *txt_top = nullptr;
+    const char *txt_bot = nullptr;
+    bool show_icon = false;
+    bool show_progress = false;
+
+    //texts
+    switch (phase_current) {
+    case PhasesSelftest::ESP_ask_auto:
+        txt_top = N_("WIFI module detected");
+        txt_bot = N_("The device firmware version missmatch. It needs to be updated.");
+        show_icon = true;
+        break;
+    case PhasesSelftest::ESP_ask_from_menu:
+        txt_top = N_("New version of WIFI firmware detected");
+        txt_bot = N_("The device firmware version missmatch. It needs to be updated.");
+        show_icon = true;
+        break;
+    case PhasesSelftest::ESP_upload:
+        txt_top = N_("WIFI update");
```

Comment:
Do we really perform a "WIFI update"? Isn't it more like an "ESP update"?

Reply:
we use esp as wi-fi module